### PR TITLE
Check if nightlies have succeeded recently enough

### PR DESCRIPTION
**Pull request by vyasr**
## Description

Contributes to https://github.com/rapidsai/build-planning/issues/127

Relies on https://github.com/rapidsai/shared-actions/pull/32

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: .github/workflows/pr.yaml
Comment (bdice) [inline .github/workflows/pr.yaml:61]: Could we use the default image then?

Comment (vyasr) [reply .github/workflows/pr.yaml]: From offline discussions, Bradley's suggestion is to use one of the images from https://github.com/actions/runner-images, specifically probably ubuntu-latest, so I'm switching to that.

Comment (vyasr) [reply .github/workflows/pr.yaml]: I had to use 24.04 to get a feature I need from Python 3.11. We can change to ubuntu-latest once that is 24.04 (or later).

Comment (bdice) [reply .github/workflows/pr.yaml]: What feature? That's fine for now.

Comment (vyasr) [reply .github/workflows/pr.yaml]: https://docs.python.org/3/library/datetime.html#datetime.date.fromisoformat didn't support timezones before 3.11.

**Comment by vyasr** [comment]
Once the PR is approved I'll change the repo name from cugraph to cudf, merge the shared-actions PR and remove the branching here, then rerun CI.

**Review by bdice — APPROVED** [review]
This is good! I also approved https://github.com/rapidsai/shared-actions/pull/32.

**Comment by vyasr** [comment]
Well, we know this works since it immediately picked up that our 25.02 nightlies are failing because of https://github.com/rapidsai/cudf/pull/17611 and https://github.com/rapidsai/cudf/pull/17612! I can merge this tomorrow once those are all handled.

**Pull request merged**
